runtime/src/lib.rs

```diff
@@ -569,13 +569,18 @@ parameter_types! {
     pub HistoryDepth: u32 = 30; // Number of eras to keep in history. Older eras cannot be claimed.
 }
 
-/// Maximum number of election targets (eligible authorities) to account for
-pub const MAX_TARGETS: u32 = 32;
-/// Maximum number of election voters to account for
-pub const MAX_VOTERS: u32 = 32;
+// Maximum number of election targets (eligible authorities) to account for. The staking pallet
+// can never have more validators than this.
+pub const MAX_TARGETS: u32 = 1_000;
+// Maximum number of voters. This also includes targets, which implicitly vote for themselves.
+pub const MAX_VOTERS: u32 = 5_000;
 
 parameter_types! {
-    pub ElectionBoundsOnChain: ElectionBounds = ElectionBoundsBuilder::default().voters_count((MAX_TARGETS + MAX_VOTERS).into()).targets_count(MAX_TARGETS.into()).build();
+    // Maximum number of election voters and targets that can be handled by OnChainSeqPhragmen
+    // in a single block time.
+    pub ElectionBoundsOnChain: ElectionBounds = ElectionBoundsBuilder::default().voters_count((MAX_VOTERS).into()).targets_count(MAX_TARGETS.into()).build();
+    // Maximum number of election winners, and thus of authorities that can be active in a given
+    // era.
     pub const MaxActiveValidators: u32 = MAX_TARGETS;
```

Comment:
Is it always true that the maximum number of active validators equals the maximum number of eligible targets ? Are there bounds enforced somewhere ?

Reply:
Yes, that's enforced by `DataProvider` configuration (staking pallet) that provide max `MAX_TARGET` targets on `desired_targets()` call.